code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -219,17 +219,22 @@
 		return
 
 	if(W.GetID())
-		if((mecha_flags & ADDING_ACCESS_POSSIBLE) || (mecha_flags & ADDING_MAINT_ACCESS_POSSIBLE))
-			if(internals_access_allowed(user))
-				ui_interact(user)
-				return
-			to_chat(user, span_warning("Invalid ID: Access denied."))
+		if(!allowed(user))
+			if(mecha_flags & ID_LOCK_ON)
+				to_chat(user, span_warning("Invalid ID: Access denied."))
+			else
+				to_chat(user, span_warning("Invalid ID: Unable to set ID lock."))
 			return
-		to_chat(user, span_warning("Maintenance protocols disabled by operator."))
+		mecha_flags ^= ID_LOCK_ON
+		to_chat(user, span_warning("ID lock [mecha_flags & ID_LOCK_ON ? "enabled" : "disabled"]."))
+		return
+
+	if(is_wire_tool(W) && (mecha_flags & PANEL_OPEN))
+		wires.interact(user)
 		return
 
 	if(istype(W, /obj/item/stock_parts/cell))
-		if(construction_state == MECHA_OPEN_HATCH)
+		if(mecha_flags & PANEL_OPEN)
```

Comment:
This if condition is repeated a lot of times, Try writing an early return for it for e.g.

```
if(istype(W, obj/item/stock_parts))
  //If we are trying to insert a stock part, don't bother if the panel isn't open
  if(!(mecha_flags & PANEL_OPEN))
     return
  .
  //check for particular type of stock part and do the regular stuff
```
Then you will have less repeated code

Reply:
That will make it impossible to attach with anything while not in combat mode and the panel is closed. And everything in `return ..()` won't work if the panel is closed. I can reduce the nesting, but the check has to stay there, I believe.

Reply:
Whatever you can do to reduce the code then,  go ahead

Reply:
Moved into a new proc try_insert_part and checking the panel inside

Reply:
Looks good. You can resolve this